src/index.d.ts

```diff
@@ -191,6 +191,8 @@ declare namespace preact {
 		diff?(vnode: VNode): void;
 		/** Attach a hook that is invoked before a vnode has rendered. */
 		render?(vnode: VNode): void;
+		/** Attach a hook that is invoked before a hook's state is queried. */
+		preHook?(component: VNode): void;
```

Comment:
```suggestion
		preHook?(component: Component): void;
```

The type of the argument here isn't a `VNode`.